pkg/pipeline/pipelines.go

```diff
@@ -0,0 +1,53 @@
+package pipeline
+
+// Function defines the general form of a pipeline function.
+type Function func(PipelineObject, *ExecutionContext) ExecutionResult
+
+type Step struct {
+	Name string
+	F    Function
+}
+
+func ReconcileTenant(obj PipelineObject, data *ExecutionContext) error {
+	steps := []Step{
+		{Name: "tenant specific steps", F: tenantSpecificSteps},
+		{Name: "create git repo", F: createGitRepo},
+		{Name: "set gitrepo url and hostkeys", F: setGitRepoURLAndHostKeys},
+		{Name: "common", F: common},
+	}
+
+	return RunPipeline(obj, data, steps)
+}
+
+func ReconcileCluster(obj PipelineObject, data *ExecutionContext) error {
+	steps := []Step{
+		{Name: "cluster specific steps", F: clusterSpecificSteps},
+		{Name: "create git repo", F: createGitRepo},
+		{Name: "set gitrepo url and hostkeys", F: setGitRepoURLAndHostKeys},
+		{Name: "add tenant label", F: addTenantLabel},
+		{Name: "common", F: common},
+	}
+
+	return RunPipeline(obj, data, steps)
+}
+
+func ReconcileGitRep(obj PipelineObject, data *ExecutionContext) error {
+	steps := []Step{
+		{Name: "deletion check", F: checkIfDeleted},
+		{Name: "git repo specific steps", F: gitRepoSpecificSteps},
+		{Name: "add tenant label", F: addTenantLabel},
+		{Name: "common", F: common},
+	}
+
+	return RunPipeline(obj, data, steps)
+}
+
+func RunPipeline(obj PipelineObject, data *ExecutionContext, steps []Step) error {
+	for _, step := range steps {
+		if r := step.F(obj, data); resultNotOK(r) {
```

Comment:
Do we still need the `resultNotOK` and `wrapError` abstractions? If I am not mistaken, they are only used here. If this is the case, drop them and do it within this function.